src/go/rpk/pkg/cli/topic/describe_storage.go

```diff
@@ -0,0 +1,258 @@
+				if b {
+					sections++
+				}
+			}
+			withSections := sections > 1
+
+			header("SUMMARY", summary, withSections, func() {
+				tw := out.NewTabWriter()
+				defer tw.Flush()
+				tw.PrintColumn("NAME", topic)
+				if topicMeta.IsInternal {
+					tw.PrintColumn("INTERNAL", topicMeta.IsInternal)
+				}
+				tw.PrintColumn("PARTITIONS", len(topicMeta.Partitions))
+				if len(topicMeta.Partitions) > 0 {
+					p0 := &topicMeta.Partitions[0]
+					tw.PrintColumn("REPLICAS", len(p0.Replicas))
+				}
+				// We can safely pick the value of the first partition since
+				// the cloud storage mode doesn't change between partitions of
+				// the same topic.
+				tw.PrintColumn("CLOUD-STORAGE-MODE", report[0].CloudStatus.CloudStorageMode)
+
+				lastUpload := getLastUpload(report)
+				tw.PrintColumn("LAST-UPLOAD", humanReadable(lastUpload, human, "time"))
+			})
+
+			header("LOG-SIZE", size, withSections, func() {
+				tw := out.NewTable("PARTITION", "TOTAL", "CLOUD", "LOCAL")
+				defer tw.Flush()
+				for _, r := range report {
+					tw.Print(
+						r.Partition,
+						humanReadable(r.CloudStatus.TotalLogSize, human, "size"),
+						humanReadable(r.CloudStatus.CloudLogSize, human, "size"),
+						humanReadable(r.CloudStatus.LocalLogSize, human, "size"),
+					)
+				}
+			})
+
+			header("SYNC", sync, withSections, func() {
+				isRrr := report[0].CloudStatus.CloudStorageMode == "read_replica"
+				headers := []string{"PARTITION", "LAST-SEGMENT-UPLOAD", "LAST-MANIFEST-UPLOAD", "METADATA-UPDATE-PENDING"}
+				if isRrr {
+					headers = append(headers, "LAST-MANIFEST-SYNC")
+				}
+				tw := out.NewTable(headers...)
+				defer tw.Flush()
+				for _, r := range report {
+					row := []any{
+						r.Partition,
+						humanReadable(r.CloudStatus.LastSegmentUpload, human, "time"),
+						humanReadable(r.CloudStatus.LastManifestUpload, human, "time"),
+						r.CloudStatus.MetadataUpdatePending,
+					}
+					// If isRrr is true, LastManifestSync should be there, this
+					// nil check is possibly redundant, but we want to avoid a
+					// nil pointer dereference.
+					if isRrr && r.CloudStatus.LastManifestSync != nil {
+						row = append(row, humanReadable(*r.CloudStatus.LastManifestSync, human, "time"))
+					}
+					tw.Print(row...)
+				}
+			})
+
+			header("LOG-SEGMENT-COUNT", count, withSections, func() {
+				tw := out.NewTable("PARTITION", "CLOUD", "LOCAL")
+				defer tw.Flush()
+				for _, r := range report {
+					tw.Print(r.Partition, r.CloudStatus.CloudLogSegmentCount, r.CloudStatus.LocalLogSegmentCount)
+				}
+			})
+
+			header("LOG-OFFSET", offset, withSections, func() {
+				tw := out.NewTable("PARTITION", "CLOUD-START", "CLOUD-LAST", "LOCAL-START", "LOCAL-LAST")
+				defer tw.Flush()
+				for _, r := range report {
+					tw.Print(r.Partition, r.CloudStatus.CloudLogStartOffset, r.CloudStatus.CloudLogLastOffset, r.CloudStatus.LocalLogStartOffset, r.CloudStatus.LocalLogLastOffset)
+				}
+			})
+		},
+	}
+	p.InstallAdminFlags(cmd)
```

Comment:
I think this needs your InstallSASLFlags from the other PR, right?

I think we need to always install sasl flags when installing admin flags, and then have a separate InstallKafkaAndAdminFlags that only installs sasl once for the few commands that install both

Reply:
Yes, you are right.

We either need AdminFlags + SASL or Admin + Kafka.

I tried adding the SASL flags into InstallAdmin but that causes cobra to panic when you install a flag twice